app/src/main/java/com/blueprint/featureA/ui/FeatureAComponent.kt

```diff
@@ -29,61 +30,90 @@ import com.blueprint.featureA.domain.model.FeatureA
 
 @Composable
 fun FeatureAScreen(viewModel: FeatureAViewModel) {
-    Scaffold { innerPadding ->
-        viewModel.getFeaturesA()
-        val features = viewModel.features.collectAsStateWithLifecycle()
-        Column(
-            modifier = Modifier
-                .fillMaxSize()
-                .padding(innerPadding)
-        ) {
-            Text(
-                "Welcome",
-                modifier = Modifier
-                    .padding(30.dp)
-            )
-            SpaceList(features = features.value) { }
-        }
-    }
+  val features by viewModel.observableFeatures.collectAsStateWithLifecycle()
+
+  FeaturesAContent(features)
 }
 
 @Composable
-fun SpaceList(features: List<FeatureA>, onCLick: () -> Unit) {
-    Box(
-        modifier = Modifier
-            .fillMaxWidth()
-            .height(200.dp)
+private fun FeaturesAContent(features: FeatureAUiState) {
+  Scaffold { innerPadding ->
+    Column(
+      modifier = Modifier
+        .fillMaxSize()
+        .padding(innerPadding)
     ) {
+      Text(
+        "Welcome",
+        modifier = Modifier
+          .padding(30.dp)
+      )
+      FeatureAList(features = features) { }
+    }
+  }
+}
+
+@Composable
+fun FeatureAList(features: FeatureAUiState, onCLick: () -> Unit) {
```

Comment:
nazvala bih ovaj arg ipak `uiState` ili slicno. Ispod fetchujes `features.features` pa je to malo cudno.

Reply:
👍